runner/lib/helpers/buffer-line-transform.js

```diff
@@ -0,0 +1,95 @@
+/* global Buffer */
+/* eslint-disable no-underscore-dangle,no-nested-ternary,no-plusplus */
+
+import { Transform } from 'stream';
+
+export default class BufferLineTransform extends Transform {
+  /**
+   * The BufferLineTransform is reading String or Buffer content from a Readable stream
+   * and writing each line as a Buffer in object mode
+   *
+   * @param {import('./buffer-line-transform.js').BufferLineTransformOptions} [options]
+   */
+  constructor(options) {
+    const { break: breakValue, breakEncoding, ...transformOptions } =
+      options || {};
+    super({ ...transformOptions, readableObjectMode: true });
+    this._breakValue = breakValue || 10;
+    this._breakEncoding = breakEncoding;
+
+    /** @type {number} */
+    let breakLength;
+    if (!breakValue || typeof breakValue === 'number') {
+      breakLength = 1;
+    } else if (Buffer.isBuffer(breakValue)) {
+      breakLength = breakValue.length;
+    } else {
+      breakLength = Buffer.from(breakValue, breakEncoding).length;
+    }
+    this._breakLength = breakLength;
+
+    /** @type {Array<Buffer>} */
+    this._chunks = [];
+  }
+
+  /**
+   * @param {any} chunk
+   * @param {BufferEncoding | 'buffer'} encoding
+   * @param {import('stream').TransformCallback} cb
+   * */
+  _transform(chunk, encoding, cb) {
+    try {
+      /** @type {Buffer} */
+      let buf =
+        Buffer.isBuffer(chunk) || encoding === 'buffer'
+          ? chunk
+          : Buffer.from(chunk, encoding);
+
+      while (buf.length) {
+        const offset = buf.indexOf(this._breakValue, 0, this._breakEncoding);
```

Comment:
Do you account for the possibility of a long break that straddles chunks?

Reply:
Good point, I do not.

Reply:
Fixed by pre-concatenating the buffers. I agree it's suboptimal, but it was that or rewriting the search algorithm in javascript to handle buffer boundaries. I wish the native node `indexOf` method had to option to report if it found a partial match at the end, as that would have made cross boundary search a lot more efficient.